dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerMemoryStore.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Memory;
+
+namespace Microsoft.SemanticKernel.Connectors.SqlServer;
+
+/// <summary>
+/// An implementation of <see cref="IMemoryStore"/> backed by a SQL Server or Azure SQL database.
+/// </summary>
+public class SqlServerMemoryStore : IMemoryStore
+{
+    internal const string DefaultSchema = "dbo";
+
+    private readonly ISqlServerClient _sqlServerClient;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqlServerMemoryStore"/> class.
+    /// </summary>
+    /// <param name="connectionString">Database connection string.</param>
+    /// <param name="schema">Database schema of collection tables.</param>
+    public SqlServerMemoryStore(string connectionString, string schema = DefaultSchema)
+        : this(new SqlServerClient(connectionString, schema))
+    { }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqlServerMemoryStore"/> class.
+    /// </summary>
+    /// <param name="sqlServerClient">An instance of <see cref="ISqlServerClient"/>.</param>
+    public SqlServerMemoryStore(ISqlServerClient sqlServerClient)
+    {
+        this._sqlServerClient = sqlServerClient;
+    }
+
+    /// <inheritdoc/>
+    public async Task CreateCollectionAsync(string collectionName, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(collectionName);
+
+        await this._sqlServerClient.CreateTableAsync(collectionName, cancellationToken).ConfigureAwait(false);
```

Comment:
FWIW not sure why we need the separation between SqlServerMemoryStore and SqlServerClient, i.e. why the code can't be directly in here... But I know other connectors are structured this way and it's not important.

Reply:
@roji In some cases, we use vector DB SDK directly, which leaves memory store implementation pretty clean, and in other cases we implement the functionality from scratch. Having `MemoryStore` and `Client` just keeps all connectors consistent. It also makes it more testable, users can implement their own `Client` if needed, while `MemoryStore` just works as an adapter.